### PR TITLE
add new package: py-pylatex

### PR DESCRIPTION
**Pull request by Sinan81**
add new package.

### CODE REVIEW

**Comment by Sinan81** [comment]
build fails with `py-setuptools@58`

**Review by tldahlgren — COMMENTED** [review]
Confirmed sha256.

File: var/spack/repos/builtin/packages/py-pylatex/package.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+class PyPylatex(PythonPackage):
+    """A Python library for creating LaTeX files and snippets"""
+
+    homepage = "https://github.com/JelteF/PyLaTeX"
+    pypi = "PyLaTeX/PyLaTeX-1.4.1.tar.gz"
+
+    version("1.4.1", sha256="d3c12efb8b260771260443dce78d1e9089c09f9d0b92e6273dfca0bf5e7302fb")
+
+    depends_on("py-setuptools", type="build")
+    depends_on("py-setuptools@:57.99", type="build", when='@:1.4.1')
```
Comment (Sinan81) [inline var/spack/repos/builtin/packages/py-pylatex/package.py]: Is it better to use a `conflict` statement here?

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-pylatex/package.py]: Depends is fine, but you should use `@:57`

File: var/spack/repos/builtin/packages/py-pylatex/package.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pylatex/package.py:34]: Missing dep on ordered-set and Python `@2.7,3.3:`

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-pylatex/package.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    homepage = "https://github.com/JelteF/PyLaTeX"
+    pypi = "PyLaTeX/PyLaTeX-1.4.1.tar.gz"
+
+    version("1.4.1", sha256="d3c12efb8b260771260443dce78d1e9089c09f9d0b92e6273dfca0bf5e7302fb")
+
+    variant('docs', default=False)
+    variant('matrices', default=False)
+    variant('matplotlib', default=False)
+    variant('quantities', default=False)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pylatex/package.py]: Needs descriptions

File: var/spack/repos/builtin/packages/py-pylatex/package.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+    depends_on("python@2.7,3.3:", type=("build", 'run')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pylatex/package.py]:
```suggestion
    depends_on("python@2.7,3.3:", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-pylatex/package.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    depends_on("py-setuptools", type="build")
+    depends_on("py-setuptools@:57", type="build", when='@:1.4.1')
+    depends_on('py-ordered-set', type='run')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pylatex/package.py]:
```suggestion
    depends_on("py-ordered-set", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-pylatex/package.py [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pylatex/package.py:26]: Missing conditional dep on future

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-pylatex/package.py (hunk 3 above)
Comment (Sinan81) [reply var/spack/repos/builtin/packages/py-pylatex/package.py]: A description isn't provided by the author o the package.

File: var/spack/repos/builtin/packages/py-pylatex/package.py [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    variant('quantities', default=False, description='Build with quantities support')
+    variant('convert_to_py2', default=False, description='Build with future support')
+
+    depends_on("python@2.7,3.3:", type=("build", 'run') )
+    depends_on("py-setuptools", type="build")
+    depends_on("py-setuptools@:57", type="build", when='@:1.4.1')
+    depends_on("py-ordered-set", type=("build", "run"))
+
+    # from extras section in setup.py
+    depends_on("py-future@0.15.2:", type=("build", 'run'), when='+convert_to_py2')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pylatex/package.py]:
```suggestion
    depends_on("py-future@0.15.2:", type=("build", 'run'), when='^python@:2')
```
This isn't optional, it's required

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-pylatex/package.py [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+    homepage = "https://github.com/JelteF/PyLaTeX"
+    pypi = "PyLaTeX/PyLaTeX-1.4.1.tar.gz"
+
+    version("1.4.1", sha256="d3c12efb8b260771260443dce78d1e9089c09f9d0b92e6273dfca0bf5e7302fb")
+
+    variant('docs', default=False, description='Build with Sphinx support for documentation')
+    variant('matrices', default=False, description='Build with matrix support')
+    variant('matplotlib', default=False, description='Build with matplotlib support')
+    variant('quantities', default=False, description='Build with quantities support')
+    variant('convert_to_py2', default=False, description='Build with future support')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pylatex/package.py]: You can remove this variant now

**Review by adamjstewart — DISMISSED** [review]

**Comment by adamjstewart** [comment]
@spackbot fix style

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**